Data/HashMap/Base.hs

```diff
@@ -385,8 +399,12 @@ equalKeys = go
       = bm1 == bm2 && A.sameArray1 go ary1 ary2
     go (Leaf h1 l1) (Leaf h2 l2) = h1 == h2 && leafEq l1 l2
     go (Full ary1) (Full ary2) = A.sameArray1 go ary1 ary2
-    go (Collision h1 ary1) (Collision h2 ary2)
-      = h1 == h2 && isPermutationBy leafEq (A.toList ary1) (A.toList ary2)
+    go (Collision h1 (L k11 _) (L k12 _) hm1) (Collision h2 (L k21 _) (L k22 _) hm2)
+      = h1 == h2 &&
+        (length hm1 == length hm2) &&
+        isPermutationBy (==)
+          (k11 : k12 : keys hm1)
+          (k21 : k22 : keys hm2)
```

Comment:
Here too, the values are illegitimately ignored.

Reply:
Same as above. This function is called `equalKeys`.